CRM/Contact/BAO/GroupContactCache.php

```diff
@@ -508,9 +508,7 @@ public static function load(&$group, $force = FALSE) {
       if (empty($contactQuery['select']) || empty($contactQuery['from'])) {
         continue;
       }
-      if (CRM_Core_DAO::singleValueQuery("SELECT COUNT(*) {$contactQuery['from']}") > 0) {
-        CRM_Core_DAO::executeQuery("INSERT IGNORE INTO $tempTable (group_id, contact_id) {$contactQuery['select']} {$contactQuery['from']}");
-      }
+      CRM_Core_DAO::executeQuery("INSERT IGNORE INTO $tempTable (group_id, contact_id) {$contactQuery['select']} {$contactQuery['from']}");
```

Comment:
I removed the `SELECT COUNT` for 2 reasons:

1. It was causing the API query to crash because swapping the `SELECT` clause  for `SELECT COUNT(*)` loses our field alias which the `HAVING` clause relies on.
2. I'm unclear about the effectiveness of this optimization. It looks to me like it's causing the query to be run twice, so I'm not sure how that's more efficient. Looks like fairly recent work of @seamuslee001 though so I'll let him chime in here and tell me why I'm wrong :)

Reply:
It looks like this actually came from @mattwire in #15588.
Any thoughts Matthew?

Reply:
@colemanw Can you pull this bit out into a separate PR? It was part of the refactoring to improve performance / reduce database deadlocks work so changing it carries quite a high risk (ie. needs testing thoroughly). That said, the refactoring was an improvement rather than an end-point so I'm happy to see further improvements like this :-)
For example, in theory we should not have needed the `INSERT IGNORE` - `INSERT` should have been ok, but in practise it wasn't.. (probably because further improvement is required to reduce concurrency).

Reply:
@mattwire sure - see #17011